etl/assets.py

```diff
@@ -3,64 +3,60 @@
 from dagster import asset
 from langchain_core.vectorstores import VectorStore
 
-from etl.embedding_model_pipelines import OpenAiEmbeddingModelPipeline
-from etl.generative_model_pipelines import OpenAiGenerativeModelPipeline
-from etl.models import (
-    DocumentTuple,
+from etl.models import DocumentTuple, RecordTuple
+from etl.pipelines import OpenAiEmbeddingPipeline, OpenAiRecordEnrichmentPipeline
+from etl.readers import WikipediaReader
+from etl.resources import (
+    InputDataFilesConfig,
     OpenAiPipelineConfig,
     OpenAiSettings,
-    RecordTuple,
-    data_files_config_from_env_vars,
-    output_config_from_env_vars,
+    OutputConfig,
 )
-from etl.readers import WikipediaReader
-from etl.utils import create_documents
 
 
 @asset
-def wikipedia_articles_from_storage() -> RecordTuple:
+def wikipedia_articles_from_storage(
+    input_data_files_config: InputDataFilesConfig,
+) -> RecordTuple:
     """Materialize an asset of Wikipedia articles."""
 
     return RecordTuple(
         records=tuple(
-            WikipediaReader(data_files_config=data_files_config_from_env_vars).read()
+            WikipediaReader(
+                data_file_paths=input_data_files_config.parse().data_file_paths
+            ).read()
         )
     )
 
 
 @asset
 def wikipedia_articles_with_summaries(
-    wikipedia_articles_from_storage: RecordTuple, config: OpenAiPipelineConfig
+    wikipedia_articles_from_storage: RecordTuple,
+    openai_pipeline_config: OpenAiPipelineConfig,
 ) -> RecordTuple:
     """Materialize an asset of Wikipedia articles with summaries."""
 
     return RecordTuple(
         records=tuple(
-            OpenAiGenerativeModelPipeline(config).enrich_record(wikipedia_article)
+            OpenAiRecordEnrichmentPipeline(openai_pipeline_config).enrich_record(
+                wikipedia_article
+            )
             for wikipedia_article in wikipedia_articles_from_storage.records
         )
     )
 
 
 @asset
 def wikipedia_articles_with_summaries_json_file(
-    wikipedia_articles_with_summaries: RecordTuple,
+    wikipedia_articles_with_summaries: RecordTuple, output_config: OutputConfig
 ) -> None:
     """Store the asset of Wikipedia articles with summaries as JSON."""
 
-    output_directory_path = output_config_from_env_vars.parse().directory_path
-
-    output_directory_path.mkdir(exist_ok=True)
-
-    enriched_wikipedia_output_file_path = (
-        output_directory_path / "wikipedia_articles_with_summaries.txt"
-    )
-
-    with enriched_wikipedia_output_file_path.open(
+    with output_config.parse().wikipedia_articles_with_summaries_file_path.open(
         mode="w"
-    ) as enriched_wikipedia_output_file:
+    ) as wikipedia_articles_with_summaries_file:
 
-        enriched_wikipedia_output_file.writelines(
+        wikipedia_articles_with_summaries_file.writelines(
             [
```

Comment:
Nit: I think writelines takes an Iterable, so you can pass in a generator here instead of creating a list first.